quipucords/scanner/openshift/api.py

```diff
@@ -192,9 +194,15 @@ def retrieve_cluster(self, **kwargs) -> OCPCluster:
     def retrieve_acm_metrics(self, **kwargs) -> list:
         """Retrieve metrics on acm managed clusters."""
         acm_metrics = []
-        for cluster in self._list_managed_clusters(**kwargs).items:
-            managed_cluster_metrics = self._init_managed_cluster(cluster)
-            acm_metrics.append(managed_cluster_metrics)
+        try:
```

Comment:
I might be over complicating things, but let me know what you think about it. ACS will also need to handle ResourceNotFoundError and we know this for a fact. Should we have a separate function raise this exception for us ?  I dunno if we will ever need to face a situation like this when we are dealing with "dynamic" resources, but it's worth asking. Example:
```

def handle_insert_good_name__exception(logger, configuration, api_func, api_name):
    try:
        managed_cluster = self.api_func()
    except ResourceNotFoundError:
        logger.info(
            _("OpenShift host at %(host)s does not have the f{api_name}API."),
            {"host": configuration.host},
        )
        return managed_cluster

def retrieve_acm_metrics(self, **kwargs) -> list:
    """Retrieve metrics on acm managed clusters."""
    acm_metrics = []
    for cluster in self._list_managed_clusters(**kwargs).items:
        managed_cluster_metrics = handle_exception(params)
        acm_metrics.append(managed_cluster_metrics)
    return acm_metrics
```

Reply:
Is there a rule of thumb to decide when we should abstract an implementation a little more ?